### PR TITLE
perf(transform): skip parsing files without `await`

### PR DESCRIPTION
**Pull request by TheAlexLichter**
Thightens `shouldTransform` given that `await` is needed as keyword.

## Summary by CodeRabbit

* **Bug Fixes**
  * Tightened transformation eligibility so code is only processed when it includes both a supported target function call and a real `await` keyword.
  * Avoids triggering on lookalike or unrelated patterns that merely contain “await” text.
* **Tests**
  * Expanded `shouldTransform` test coverage with additional negative and edge-case inputs (e.g., target call without `await`, `await` without the target call, and non-keyword “await” usage).

### CODE REVIEW

File: src/transform.ts [hunk 1]
```diff
@@ -66,7 +68,9 @@ export function createTransformer(options: TransformerOptions = {}): {
   );
 
   function shouldTransform(code: string): boolean {
-    return typeof code === "string" && matchRE.test(code);
+    return (
+      typeof code === "string" && matchRE.test(code) && AWAIT_RE.test(code)
```
Comment (pi0) [inline src/transform.ts:72]: I think we can reorder await check before matchRe as it is more predictable/restrictive.

Comment (TheAlexLichter) [reply src/transform.ts:72]: I don't think that'll be more efficient.

At the moment, we check less files for both the functions + await
If we put the await regex first we have more files in the first selection (more files have await than one of the functions checked `matchRE`), and *then* reject a lot of files after.

**Comment by TheAlexLichter** [comment]
CI is failing due to setup issues that I resolved via #134 already

**Review by pi0 — APPROVED** [review]
